test/test-lex.cpp

```diff
@@ -113,6 +113,149 @@ TEST_F(test_lex, lex_block_comments) {
   }
 }
 
+TEST_F(test_lex, lex_unopened_block_comment) {
+  {
+    error_collector v;
+    padded_string input(u8"hello */"_sv);
+    lexer l(&input, &v);  // identifier
+    EXPECT_EQ(l.peek().type, token_type::identifier);
+    l.skip();  // end of file
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_unopened_block_comment, comment_close,
+                    offsets_matcher(&input, strlen(u8"hello "), u8"*/"))));
+  }
+}
+
+TEST_F(test_lex, lex_unopened_block_comment1) {
+  {
+    error_collector v;
+    padded_string input(u8"*-----*/"_sv);
+    lexer l(&input, &v);
+
+    while (l.peek().type != token_type::end_of_file) {
+      l.skip();
+    }
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_unopened_block_comment, comment_close,
+                    offsets_matcher(&input, strlen(u8"*-----"), u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"*******/"_sv);
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::star_star);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::star_star);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::star_star);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_unopened_block_comment, comment_close,
+                    offsets_matcher(&input, strlen(u8"******"), u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"*/"_sv);
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
+                              error_unopened_block_comment, comment_close,
+                              offsets_matcher(&input, 0, u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"foo(a, true * direct */);"_sv);
```

Comment:
Nit: Is this test case really different from the others? I don't think this test carries its own weight. I'd delete it.